### PR TITLE
Restrict data processed by the publishing_api:publishing_api_patch_links task

### PR DESCRIPTION
**Pull request by MatMoore**
Restrict the data processed by the rake task for sending links to the publishing api.

* When running on a specific subclass, restrict to published editions
* Allow specifying a comma separated list of class names
* Added some more output to indicate progress

### CODE REVIEW

File: lib/tasks/publishing_api.rake [hunk 1]
```diff
@@ -56,19 +56,22 @@ namespace :publishing_api do
...
-    if class_name == 'Edition'
-      klass = Edition.published
-    else
-      klass = class_name.constantize
-    end
+    args[:item_classes].split(',').each do |class_name|
+      if class_name == 'Edition'
+        editions = Edition.published
+      else
+        editions = class_name.constantize.published
```
Comment (tijmenb) [inline lib/tasks/publishing_api.rake]: I'm not a 100% sure everything we will migrate has a 'published' scope - have you checked?

Comment (MatMoore) [inline lib/tasks/publishing_api.rake]: I'll check this - had assumed it was always inherited from Edition

Comment (MatMoore) [inline lib/tasks/publishing_api.rake]: The scope comes from Editions::Workflow so it should be there for everything.

Comment (tijmenb) [inline lib/tasks/publishing_api.rake]: Checked this, and the only thing on our list that doesn't have a `published` scope is `Policy`.

Comment (MatMoore) [inline lib/tasks/publishing_api.rake]: Aren't policies published by policy publisher though?

Either way, this check is not doing anything useful any more - I think it should probably just check the class descends from Edition and bail out if given anything else.

File: lib/tasks/publishing_api.rake [hunk 2]
```diff
@@ -56,19 +56,25 @@ namespace :publishing_api do
...
     end
 
-    class_name = args[:item_class]
-    if class_name == 'Edition'
-      klass = Edition.published
-    else
-      klass = class_name.constantize
-    end
+    args[:item_classes].split(',').each do |class_name|
+      cls = class_name.constantize
```
Comment (tijmenb) [inline lib/tasks/publishing_api.rake]: Can you name this `klass`? That's more idiomatic.

Comment (MatMoore) [inline lib/tasks/publishing_api.rake]: Thanks, fixed.

**Comment by tijmenb** [comment]
:+1: merge when green

**Pull request merged**